test/conformance/conformance_blocked_nd_range.cpp

```diff
@@ -291,3 +292,148 @@ TEST_CASE("blocked_nd_range proportional splitting") {
         utils::check_range_bounds_after_splitting(original.dim(0), first.dim(0), second.dim(0), expected_first_end);
     }
 }
+
+#if __TBB_CPP17_DEDUCTION_GUIDES_PRESENT
+template <typename T>
+void test_deduction_guides() {
+    using oneapi::tbb::blocked_nd_range;
+    static_assert(std::is_constructible<T, int>::value, "Incorrect test setup");
+    // T as a grainsize in braced-init-list constructions should be used since only
+    // the same type is allowed by the braced-init-list
+    static_assert(std::is_convertible<T, typename blocked_nd_range<T, 1>::size_type>::value,
+                  "Incorrect test setup");
+
+    std::vector<T> v;
+    using iterator = typename decltype(v)::iterator;
+
+    oneapi::tbb::blocked_range<T> dim_range(0, 100);
+
+    blocked_nd_range<T, 2> source_range(dim_range, dim_range);
+
+    {
+        blocked_nd_range range(dim_range, dim_range, dim_range);
+        static_assert(std::is_same_v<decltype(range), blocked_nd_range<T, 3>>);
+    }
+    {
+        blocked_nd_range range({v.begin(), v.end()}, {v.begin(), v.end()});
+        static_assert(std::is_same_v<decltype(range), blocked_nd_range<iterator, 2>>);
+    }
+    {
+        blocked_nd_range range({T{0}, T{100}}, {T{0}, T{100}, T{5}}, {T{0}, T{100}}, {T{0}, T{100}, T{5}});
+        static_assert(std::is_same_v<decltype(range), blocked_nd_range<T, 4>>);
```

Comment:
I would add a check on the grainsizes of the resulting subranges. For example, this four-dimensional `range` seems to be a good candidate for this as it includes subranges with different grainsizes.